core/interop/src/lib.rs

```diff
@@ -40,67 +46,330 @@ impl<T: IntoIterator<Item = (JsString, NativeFunction)> + Clone> IntoJsModule fo
+        }
+    }
+}
+
+/// An argument that when used in a JS function will empty the list
+/// of JS arguments as `JsValue`s. This can be used for having the
+/// rest of the arguments in a function. It should be the last
+/// argument of your function, before the `Context` argument if any.
+///
+/// For example,
+/// ```
+/// # use boa_engine::{Context, JsValue};
+/// # use boa_interop::{IntoJsFunctionCopied, JsRest};
+/// # let mut context = Context::default();
+/// let sums = (|args: JsRest, context: &mut Context| -> i32 {
+///    args.iter().map(|i| i.try_js_into::<i32>(context).unwrap()).sum::<i32>()
+/// }).into_js_function_copied(&mut context);
+///
+/// let result = sums.call(
+///     &JsValue::undefined(),
+///     &[JsValue::from(1), JsValue::from(2), JsValue::from(3)],
+///     &mut context
+/// ).unwrap();
+/// assert_eq!(result, JsValue::new(6));
+/// ```
+#[derive(Debug, Clone)]
+pub struct JsRest(pub Vec<JsValue>);
+
+#[allow(unused)]
+impl JsRest {
+    /// Consumes the `JsRest` and returns the inner list of `JsValue`.
+    #[must_use]
+    pub fn into_inner(self) -> Vec<JsValue> {
+        self.0
+    }
+
+    /// Returns an iterator over the inner list of `JsValue`.
+    pub fn iter(&self) -> impl Iterator<Item = &JsValue> {
+        self.0.iter()
+    }
+
+    /// Returns a mutable iterator over the inner list of `JsValue`.
+    pub fn iter_mut(&mut self) -> impl Iterator<Item = &mut JsValue> {
+        self.0.iter_mut()
+    }
+
+    /// Returns the length of the inner list of `JsValue`.
+    #[must_use]
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    /// Returns `true` if the inner list of `JsValue` is empty.
+    #[must_use]
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+}
+
+impl From<&[JsValue]> for JsRest {
+    fn from(values: &[JsValue]) -> Self {
+        Self(values.to_vec())
+    }
+}
+
+impl IntoIterator for JsRest {
+    type Item = JsValue;
+    type IntoIter = std::vec::IntoIter<JsValue>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.into_inner().into_iter()
+    }
+}
+
+/// An argument that when used in a JS function will capture all
+/// the arguments that can be converted to `T`. The first argument
+/// that cannot be converted to `T` will stop the conversion.
+///
+/// For example,
+/// ```
+/// # use boa_engine::{Context, JsValue};
+/// # use boa_interop::{IntoJsFunctionCopied, JsAll};
+/// # let mut context = Context::default();
+/// let sums = (|args: JsAll<i32>, context: &mut Context| -> i32 {
+///    args.iter().sum()
+/// }).into_js_function_copied(&mut context);
+///
+/// let result = sums.call(
+///     &JsValue::undefined(),
+///     &[JsValue::from(1), JsValue::from(2), JsValue::from(3), JsValue::Boolean(true), JsValue::from(4)],
+///     &mut context
+/// ).unwrap();
+/// assert_eq!(result, JsValue::new(6));
+/// ```
+#[derive(Debug, Clone)]
+pub struct JsAll<T: TryFromJs>(pub Vec<T>);
```

Comment:
question: does `Vec<T>` need to be `pub` here with `into_inner`.

I think it makes sense for these to be `pub(crate)` to begin unless there is a specific need for them to be `pub`.

Reply:
It's more a convenience. This way I can declare

```rust
fn my_js_fn(JsAll(all_numbers): JsAll<i32>) { ... } 
```

I'd rather remove `into_inner()` than remove the `pub`, tbh.